plugin/hook/adapter_test.go

```diff
@@ -0,0 +1,622 @@
+			Multiselect: []string{"a", "ba", "b"},
+			SubQuestions: []models.PluginSubQuestion{
+				{
+					IfValue: "",
+					Questions: []models.PluginQuestion{
+						{
+							Name:         "nil",
+							Prompt:       "nil",
+							Help:         "nil",
+							Default:      "nil",
+							Multiselect:  nil,
+							SubQuestions: nil,
+						},
+					},
+				},
+			},
+		})
+		expected := &pb.PluginQuestion{
+			Name:        "key",
+			Prompt:      "sec",
+			Help:        "this",
+			Default:     "do",
+			Multiselect: []string{"a", "ba", "b"},
+			SubQuestions: []*pb.PluginQuestion_SubQuestion{
+				{
+					IfValue: "",
+					Questions: []*pb.PluginQuestion{
+						{
+							Name:         "nil",
+							Prompt:       "nil",
+							Help:         "nil",
+							Default:      "nil",
+							Multiselect:  nil,
+							SubQuestions: nil,
+						},
+					},
+				},
+			},
+		}
+		assert.Equal(t, expected, actual)
+	})
+	t.Run("should work for non empty inputs", func(t *testing.T) {
+		actual := adaptQuestionToProto(models.PluginQuestion{
+			Name:        "key",
+			Prompt:      "sec",
+			Help:        "this",
+			Default:     "do",
+			Multiselect: []string{"a", "ba", "b"},
+			SubQuestions: []models.PluginSubQuestion{
+				{
+					IfValue: "",
+					Questions: []models.PluginQuestion{
+						{
+							Name:         "key",
+							Prompt:       "sec",
+							Help:         "this",
+							Default:      "do",
+							Multiselect:  []string{"a", "ba", "b"},
+							SubQuestions: []models.PluginSubQuestion{},
+						},
+					},
+				},
+			},
+		})
+		expected := &pb.PluginQuestion{
+			Name:        "key",
+			Prompt:      "sec",
+			Help:        "this",
+			Default:     "do",
+			Multiselect: []string{"a", "ba", "b"},
+			SubQuestions: []*pb.PluginQuestion_SubQuestion{
+				{
+					IfValue: "",
+					Questions: []*pb.PluginQuestion{
+						{
+							Name:         "key",
+							Prompt:       "sec",
+							Help:         "this",
+							Default:      "do",
+							Multiselect:  []string{"a", "ba", "b"},
+							SubQuestions: nil,
+						},
+					},
+				},
+			},
+		}
+		assert.Equal(t, expected, actual)
+	})
+}
+
+func Test_adaptQuestionFromProto(t *testing.T) {
+	t.Run("should work for empty input", func(t *testing.T) {
+		actual := adaptQuestionFromProto(nil)
+		expected := models.PluginQuestion{}
+		assert.Equal(t, expected, actual)
+	})
+	t.Run("should work for non empty inputs (with empty questions parameter)", func(t *testing.T) {
+		actual := adaptQuestionFromProto(&pb.PluginQuestion{
+			Name:         "key",
+			Prompt:       "sec",
+			Help:         "this",
+			Default:      "do",
+			Multiselect:  nil,
+			SubQuestions: nil,
+		})
+		expected := models.PluginQuestion{
+			Name:         "key",
+			Prompt:       "sec",
+			Help:         "this",
+			Default:      "do",
+			Multiselect:  nil,
+			SubQuestions: []models.PluginSubQuestion{},
+		}
+		assert.Equal(t, expected, actual)
+	})
+	t.Run("should work for nil input", func(t *testing.T) {
```

Comment:
This test and the test at 168 is same. In Go, `"nil"` is not equal to `nil`

Reply:
Is `SubQuestions: nil,` can be written as `SubQuestions: []*pb.PluginQuestion_SubQuestion{},` ?